src/Reflection/AST/DeBruijn.agda

```diff
@@ -53,6 +55,16 @@ module _ where
   weakenClauses : (by : ℕ) → Clauses → Clauses
   weakenClauses = weakenFrom′ traverseClauses 0
 
+-- Apply Weakening to substitute under lambdas
+
+prependLams : List (String × Visibility) → Term → Term
+prependLams xs t = foldr (λ {(s , v) t → lam v (abs s t)}) t (reverse xs)
```

Comment:
Note that you don't need a pattern-matching lambda for irrefutable patterns like `(s , v)`

Reply:
Thanks! Didn't know that..